code/mob/living/critter/ai/flock/flocktasks.dm

```diff
@@ -757,33 +775,21 @@ butcher
 	var/mob/living/critter/flock/drone/F = holder.owner
 	if(!F?.flock)
 		return
-	for(var/atom/T in view(target_range, holder.owner))
-		//handle vehicles first
-		if (isvehicle(T))
-			var/vehicle_is_enemy = F.flock.isEnemy(T) //if someone gets in an enemy pod, attack
-			var/enemy_found = FALSE
-			for (var/mob/occupant in T)
-				if (!F.flock.isEnemy(occupant) && !vehicle_is_enemy)
+	for(var/atom/T in F.flock.enemies)
+		if(T in view(holder.owner,target_range))
+			F.flock.updateEnemy(T)
+			if(isliving(T))
+				var/mob/living/M = T
+				if(is_incapacitated(M))
 					continue
-				F.flock.updateEnemy(occupant)
-				enemy_found = TRUE
-				break
-			if (enemy_found) //bad guy in pod
-				. += T
-			//continue regardless of whether we find an enemy or not, since we don't want to attack empty pods
-			continue
-		if(!F.flock.isEnemy(T))
-			continue
-		if(isliving(T))
-			var/mob/living/M = T
-			if(is_incapacitated(M))
+			if(istype(T.loc, /obj/flock_structure/cage))
```

Comment:
For T.loc, cage is checked here, but vehicle is checked for outside?

Reply:
vehicles can't be in cages